src/uct/cuda/cuda_copy/cuda_copy_ep.c

```diff
@@ -97,6 +98,40 @@ uct_cuda_copy_get_mem_type(uct_md_h md, void *address, size_t length)
     return mem_info.type;
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t uct_cuda_copy_ctx_rsc_get(
+        uct_cuda_copy_iface_t *iface, uct_cuda_copy_ctx_rsc_t **ctx_rsc_p)
+{
+    CUcontext ctx;
+    ucs_status_t status;
+    khiter_t iter;
+    unsigned long long ctx_id;
+
+    status = UCT_CUDADRV_FUNC_LOG_ERR(cuCtxGetCurrent(&ctx));
+    if (ucs_unlikely(status != UCS_OK)) {
+        return status;
+    } else if (ucs_unlikely(ctx == NULL)) {
+        ucs_error("no context bound to calling thread");
```

Comment:
1. "no cuda context bound to calling thread"
2. didn't we decide in this case to use primary context of device 0?